ccdproc/shutter_correction.py

```diff
@@ -0,0 +1,347 @@
+from .image_collection import ImageFileCollection
+
+
+def GaladiEnriqez1995(combined_flats, exptimekey='EXPTIME', output=None):
+    '''
+    Determine the shutter correction map given a set of bias subtracted flats
+    taken under constant illumination.
+    
+    Parameters
+    ----------
+    combined_flats : `list` containing `~ccdproc.CCDData` objects
+        Set of bias subtracted flats of different exposure times taken under
+        constant illumination.
+    
+    exptimekey : `string`, optional, defaults to 'EXPTIME'
+        The header keyword which contains the exposure time of the image in
+        seconds.
+
+    output : `str`, optional
+        The filename to write the output shutter map to.  This will be a FITS
+        file with two extensions.  The first contains the shutter map values
+        (in seconds) and the second contains the uncertainty in those values.
+
+    Notes
+    -----
+    This code follows the implementation described in Galadi-Enriquez, D.,
+    Jordi, C., & Trullols, E. (1995). "Effects of Shutter Timing on CCD
+    Photometry". IAU Symposium No. 167, 167, 327.
+
+    Returns
+    -------
+    shutter_map : `~ccdproc.CCDData`
+        The shutter correction map in units of seconds.
+    '''
+    n = len(combined_flats)
+    exptimes = np.ma.MaskedArray([f.header[exptimekey] for f in combined_flats],
+                                 mask=np.zeros(n, dtype=bool))
+    wmax = np.argmax(exptimes)
+    exptimes.mask[wmax] = True
+    cflats = np.array([f.data for i,f in enumerate(combined_flats) if i != wmax])
+    R = np.sum(cflats, axis=0) / combined_flats[wmax].data
+    delta = (exptimes.data[wmax]*R - exptimes.sum()) / (n-1 - R)
+    shutter_map = CCDData(delta, unit=u.second)
+    if output:
+        if os.path.exists(output):
+            os.remove(output)
+        fits_ccddata_writer(shutter_map, output)
+    return shutter_map
+
+
+def apply_shutter_map(image, shutter_map, exptimekey='EXPTIME'):
+    '''
+    Correct an input image using the given shutter map.
+    
+    Parameters
+    ----------
+    image : `~ccdproc.CCDData`
+        The image to be corrected.
+
+    shutter_map : `~ccdproc.CCDData`
+        The shutter map to use to correct the image.
+
+    exptimekey : `string`, optional, defaults to 'EXPTIME'
+        The header keyword which contains the exposure time of the image in
+        seconds.
+
+    Returns
+    -------
+    corrected_image : `~ccdproc.CCDData`
+        The image scaled to a 1 second exposure.  The units will be the input
+        units per second.  For eample, if the input image is in units of adu, 
+        then the output will be in units of adu / s.
+    '''
+    exptime_map = shutter_map.add(image.header[exptimekey]*u.second)
+    corrected_image = image.divide(exptime_map)
+    return corrected_image
+
+
+def get_shutter_map(files, exptimekey='EXPTIME', min_files_to_sigma_clip=5,
+                    sigma_clip_low_thresh=5, sigma_clip_high_thresh=5,
```

Comment:
Like the comment below, this seems more like a full pipeline for reducing the data and create the bias step.  It is likely that users have already reduced the data or have reduced the data in there preferred way in a different step in the process.  I would suggest splitting these two steps  (the reduction and the creation of the shutter map) to give the users more flexibility.  However, I have no problem keeping a pipeline step like this as a convenience function.